ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDTopicTableViewController.m

```diff
@@ -0,0 +1,223 @@
+//
+//  ViewController.m
+//  ReminderApp-OBJC
+//
+//  Created by Catalin David on 01/07/16.
+//  Copyright © 2016 Catalin David. All rights reserved.
+//
+
+#import "CDTopicTableViewController.h"
+#import "AppDelegate.h"
+#import "CDTopic.h"
+#import "CDReminderTableViewController.h"
+
+#pragma mark - Class extension
+
+@interface CDTopicTableViewController () <NSFetchedResultsControllerDelegate>
+
+@property (nonatomic, weak) IBOutlet UIBarButtonItem *editButton;
+@property (nonatomic, strong) NSFetchedResultsController *fetchedResultsController;
+@property (nonatomic, strong) NSManagedObjectContext *managedContext;
+
+@end
+
+#pragma mark - Class implementation
+
+@implementation CDTopicTableViewController
+
+- (void)viewDidLoad {
+	
+	[super viewDidLoad];
+	
+	self.editButton.title = @"Edit";
+	self.title = @"Topics";
+	
+	self.managedContext = ((AppDelegate*)[UIApplication sharedApplication].delegate).managedObjectContext;
+	
+	NSError *error;
+	if (![self.fetchedResultsController performFetch:&error]) {
+		NSLog(@"An error occured: %@", error);
+	}
+	
+}
+
+-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+	
+	if ([segue.identifier isEqualToString:@"segueTopicControllerToReminderController"]) {
+		CDReminderTableViewController *destinationController = ((CDReminderTableViewController *)segue.destinationViewController);
+		destinationController.topic = ((CDTopic *) [self.fetchedResultsController objectAtIndexPath:self.tableView.indexPathForSelectedRow]);
+	}
+	
+}
+
+#pragma mark - Specific action buttons
```

Comment:
Ar merge "action methods for buttons", din specificatia ta se intelege ca sunt butoane acolo. Schimba textul.